cudamapper/include/claraparabricks/genomeworks/cudamapper/index.hpp

```diff
@@ -187,6 +187,46 @@ class IndexHostCopyBase
                                                            const cudaStream_t cuda_stream = 0);
 };
 
+/// IndexDescriptorBase - An "almost an Index" class from which an index can be created.
+/// Every Index is defined by its first read and the number of reads.
+class IndexDescriptorBase
+{
+public:
+    /// \brief destructor
+    virtual ~IndexDescriptorBase() = default;
+
+    /// \brief getter
+    virtual read_id_t first_read() const = 0;
+
+    /// \brief getter
+    virtual number_of_reads_t number_of_reads() const = 0;
+
+    /// \brief returns hash value
+    virtual std::size_t get_hash() const = 0;
+
+    /// \brief generates a mapping of (k,w)-kmer-representation to all of its occurrences for one or more sequences
+    /// \param first_read The first read for the index descriptor
+    /// \param number_of_reads The number of reads until the end of the current index
+    /// \return instance of an IndexDescriptor
+    static std::unique_ptr<IndexDescriptorBase>
+    create_index_descriptor(read_id_t first_read, number_of_reads_t number_of_reads);
```

Comment:
This function is only declared, but not defined.
Do we want users to create custom index descriptors?
So far they seem to taken from `group_reads_into_descriptors` only.

Reply:
removed in favor of exposing the IndexDescriptor class directly, as discussed previously